uae4all_gp2x_0.7.2a/src/drawing.cpp

```diff
@@ -397,37 +397,41 @@ static int src_pixel;
 static int unpainted;
 
 #define LNAME linetoscr_16
-#define SRC_INC 1
 #include "linetoscr.h"
-#undef SRC_INC
 #undef LNAME
 
-#define LNAME linetoscr_16_shrink1
-#define SRC_INC 2
+#define LNAME linetoscr_16_double
+#define HDOUBLE 1
 #include "linetoscr.h"
-#undef SRC_INC
 #undef LNAME
 
 static void pfiled_do_linetoscr_1(int start, int stop)
 {
-	src_pixel = linetoscr_16_shrink1 (src_pixel, start, stop);
+    src_pixel = linetoscr_16 (src_pixel, start, VISIBLE_LEFT_BORDER+(stop-VISIBLE_LEFT_BORDER)*2);  //mithrendal hires fix.
 }
 
 static void pfiled_do_linetoscr_0(int start, int stop)
 {
-	src_pixel = linetoscr_16 (src_pixel, start, stop);
+    src_pixel = linetoscr_16_double(src_pixel, start, stop); //mithrendal stretching of lores fix
 }
 
 static line_draw_func *pfield_do_linetoscr=(line_draw_func *)pfiled_do_linetoscr_0;
 
+
 static void pfield_do_fill_line(int start, int stop)
 {
-    register uae_u16 *b = &(((uae_u16 *)xlinebuffer)[start]);
+    //with center fix for 640pixel output. mithrendal
+    register uae_u16 *b = &(((uae_u16 *)xlinebuffer)[VISIBLE_LEFT_BORDER+(start-VISIBLE_LEFT_BORDER)*2]);
     register xcolnr col = colors_for_drawing.acolors[0];
     register int i;
     register int max=(stop-start);
-    for (i = 0; i < max; i++,b++)
-		*b = col;
+    for (i = 0; i < max; i++)
+    {
+        *b = col;
+        b++;
+        *b = col;
+        b++;
+    }
```

Comment:
This is the drawing method into the SDL buffer for amiga lores, e.g. 320 horizontal pixel. I draw one Amiga pixel "col" to two SDL pixel *b. If you would draw 1:1 pixel here, the screen would only getting drawn in the first half of the iOS screen.